### PR TITLE
Allow using btrfs and lvmpv as format for all devices

### PR DESCRIPTION
**Pull request by vojtechtrefny**
Remove special "type" for adding lvmpv and allow using lvmpv as a format for all devices.
Same applies for btrfs, but "btrfs volume" can be still added to allow adding btrfs volume with multiple (non-existing) parents.

### CODE REVIEW

File: blivetgui/blivet_utils.py [hunk 1]
```diff
@@ -783,7 +786,22 @@ def _create_format(self, user_input, device):
...
+            return [blivet.deviceaction.ActionCreateFormat(device, new_fmt)]
+
+    def _create_btrfs_format(self, user_input, device):
+        actions = []
+
+        # format the device to btrfs
+        btrfs_fmt = blivet.formats.get_format(fmt_type="btrfs")
+        actions.append(blivet.deviceaction.ActionCreateFormat(device, btrfs_fmt))
+
+        if not hasattr(user_input, "create_volume") or user_input.create_volume:
```
Comment (vpodzime) [inline blivetgui/blivet_utils.py]: ``getattr()`` would be nicer here.

File: blivetgui/dialogs/add_dialog.py [hunk 2]
```diff
@@ -834,6 +834,8 @@ def update_fs_chooser(self):
         supported_fs = supported_filesystems()
         if self.selected_free.size > size.Size("8 MiB"):
             supported_fs.append("lvmpv")
+        if self.selected_free.size > size.Size("256 MiB"):
```
Comment (vpodzime) [inline blivetgui/dialogs/add_dialog.py]: Please use some properly-named constants, ideally imported from blivet.

File: blivetgui/dialogs/add_dialog.py [hunk 3]
```diff
@@ -861,6 +863,8 @@ def on_filesystems_combo_changed(self, combo):
         dev_min_size = self._get_min_size()
         if selection == "lvmpv":
             self.update_size_area_limits(min_size=max(dev_min_size, size.Size("8 MiB")))
+        elif selection == "btrfs":
+            self.update_size_area_limits(min_size=max(dev_min_size, size.Size("256 MiB")))
```
Comment (vpodzime) [inline blivetgui/dialogs/add_dialog.py]: Same here.

**Comment by vpodzime** [comment]
Looks good to me otherwise.

**Comment by vojtechtrefny** [comment]
New version based on @vpodzime comments.

File: blivetgui/dialogs/add_dialog.py (hunk 2 above)
Comment (vpodzime) [inline blivetgui/dialogs/add_dialog.py]: There's ``MIN_MEMBER_SIZE`` in ``blivet.devicelibs.btrfs``

Comment (vpodzime) [inline blivetgui/dialogs/add_dialog.py]: And the ``8 MiB`` should be added as a constant to either blivet or somewhere here in blivet-gui. Why is it ``8 MiB`` anyway?

File: blivetgui/dialogs/add_dialog.py [hunk 4]
```diff
@@ -817,38 +817,39 @@ def add_fs_chooser(self):
...
-
         for fs in supported_fs:
-            self.filesystems_combo.append_text(fs)
+            if self.selected_free.size > fs._min_size:
+                self.filesystems_store.append((fs, fs.type, fs.name))
+        self.filesystems_store.append((None, "unformatted", _("unformatted")))
 
-        if "ext4" in supported_fs:
-            self.filesystems_combo.set_active(supported_fs.index("ext4"))
+        if "ext4" in [fs.type for fs in supported_fs]:
```
Comment (vpodzime) [inline blivetgui/dialogs/add_dialog.py]: This could be a generator instead of a list comprehension.

File: blivetgui/dialogs/edit_dialog.py [hunk 5]
```diff
@@ -121,16 +121,17 @@ def __init__(self, main_window, edit_device):
...
-            supported_fs.append("lvmpv")
         for fs in supported_fs:
-            self.fs_combo.append_text(fs)
+            if self.edit_device.size > fs._min_size:
+                self.fs_store.append((fs, fs.type, fs.name))
+        self.fs_store.append((None, "unformatted", _("unformatted")))
 
-        if "ext4" in supported_fs:
-            self.fs_combo.set_active(supported_fs.index("ext4"))
+        if "ext4" in [fs.type for fs in supported_fs]:
```
Comment (vpodzime) [inline blivetgui/dialogs/edit_dialog.py]: Same here.

File: blivetgui/dialogs/add_dialog.py (hunk 2 above)
Comment (vpodzime) [inline blivetgui/dialogs/add_dialog.py]: I see these are gone in the following commit.

**Pull request merged**